### DISCUSSION TITLE
No Homebrew ruby 2.6.8 available for arm64 processors!

### DISCUSSION THREAD

**Discussion by MrSnoozles** [Homebrew/discussions#2351] [Getting started] [answered]
Hi,

I'm trying to use homebrew on a M1 Mac mini, but most commands are not working because of the following error and I don't know how to solve it anymore:

`Error: No Homebrew ruby 2.6.8 available for arm64 processors!`
`Error: Failed to upgrade Homebrew Portable Ruby!`

This error appears on almost all brew commands, like `brew update` or `brew install <package>`. 

`brew update-reset` worked, but didn't change anything. The output of `which ruby` is `/usr/bin/ruby`, so it's just the system Ruby installed. The output of `ruby -v` is `ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.arm64e-darwin20]`
The macOS version is Big Sur (11.5.2)

Does anyone have an idea what could be causing this problem?

**Comment by carlocab** [comment, marked as answer]
Do you have `HOMEBREW_FORCE_VENDOR_RUBY` set? What's the output of 
```
ls -la /opt/homebrew/Library/Homebrew/vendor
```
?

**Reply by Opsi** [reply]
I have the exact same problem (Same Mac, exact same ruby version and exact same Big Sur version and same error output)

My `HOMEBREW_FORCE_VENDOR_RUBY` is **not** set
The output of 

```
ls -la /opt/homebrew/Library/Homebrew/vendor
```

is

```
ls: /opt/homebrew/Library/Homebrew/vendor: No such file or directory
```

**Reply by MrSnoozles** [reply]
My output of `ls -la /opt/homebrew/Library/Homebrew/vendor` is

![Screenshot 2021-10-26 at 14 07 45](https://user-images.githubusercontent.com/17496592/138875900-de5c680f-3a21-4333-89e4-f0dead33157a.png)


`HOMEBREW_FORCE_VENDOR_RUBY` is not set for me either. If I set it to 1 I still can not install packages.

**Reply by MikeMcQuaid** [reply]
@MrSnoozles @Opsi Can we see the output of `which brew`, `brew --repo` and `ls $(brew --repo)/Library/Homebrew/vendor`

**Reply by carlocab** [reply]
Please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse (e.g. when looking up error messages). Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

**Reply by Opsi** [reply]
> @MrSnoozles @Opsi Can we see the output of `which brew`, `brew --repo` and `ls $(brew --repo)/Library/Homebrew/vendor`

##### `which brew`

```
/usr/local/bin/brew
```

##### `brew --repo`

```
/usr/local/Homebrew
```

##### `ls $(brew --repo)/Library/Homebrew/vendor`

```
bundle			portable-ruby		portable-ruby-version
```

**Reply by SMillerDev** [reply]
So you're using Homebrew with Rosetta then, that's an important distinction. It shouldn't even try to use arm64 then.

CC @Bo98 @MikeMcQuaid

**Reply by MikeMcQuaid** [reply]
@Opsi You have the wrong `brew` in your `PATH`. The error message should be improved but you're trying to run the x86_64 Homebrew in an arm64 terminal.

**Reply by Opsi** [reply]
I uninstalled and reinstalled brew and now I'm using `/opt/homebrew/bin/brew` and its working. Thanks for your help.
